### PR TITLE
feat(frontend): support creating empty arrays

### PR DESCRIPTION
**Pull request by wzzzzd**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Now we can create empty arrays as follows:
```
dev=> select {};
 ?column? 
----------
 {}
(1 row)
dev=> create table t (a int[]);
CREATE_TABLE
dev=> insert into t values ({});
INSERT 0 1
dev=> select * from t;
 a  
----
 {}
(1 row)
```

This PR may violate pg's behavior. For example, in pg:
```
postgres=# create table t (a int[]);
CREATE TABLE
postgres=# insert into t values (array[]);
ERROR:  cannot determine type of empty array
LINE 1: insert into t values (array[]);
                              ^
HINT:  Explicitly cast to the desired type, for example ARRAY[]::integer[].
```

I think inserting an empty array without explicit casting is reasonable, because the array type can be inferred according to the table schema. Similarly, we can insert array[NULL] without explicit casting and this is not allowed in pg (https://github.com/singularity-data/risingwave/issues/4612).

## Checklist

- [x] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests (WIP)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: e2e_test/batch/types/array_ty.slt.part [hunk 1]
```diff
@@ -16,7 +16,7 @@ select ARRAY[null];
 ----
 {NULL}
 
-statement error
+statement ok
```
Comment (neverchanje) [inline e2e_test/batch/types/array_ty.slt.part:19]:
```
postgres=# select ARRAY[];
ERROR:  cannot determine type of empty array
LINE 1: select ARRAY[];
               ^
HINT:  Explicitly cast to the desired type, for example ARRAY[]::integer[].
```

This statement is not allowed in postgres.

**Comment by neverchanje** [comment]
I personally recommend implementing the behavior of pg, since it'd be better to minimize our feature set before we become stable enough.

**Comment by neverchanje** [comment]
It seems we don't support

```
select ARRAY[]::integer[];
```

We can start from this simple behavior. Btw I also found that `select '{}'::integer[];` will panic.

**Comment by fuyufjh** [comment]
Hi, any updates? @liurenjie1024 Can u help to take a look

**Comment by xiangjinwu** [comment]
Taking it over. It would be a trivial update after rebasing #4741

**Comment by xiangjinwu** [comment]
Replaced by #5402 as this one is `wzzzzd:support_empty_list` rather than within this repo.

**Pull request closed without merging**